crates/storage/trie/db/libmdbx.rs

```diff
@@ -1,70 +1,90 @@
+use std::marker::PhantomData;
+
 use crate::error::StoreError;
 use libmdbx::{
-    orm::{table, Database},
+    orm::{table, Database, Table},
     table_info,
 };
 
 /// Libmdbx implementation for the TrieDB trait, with get and put operations.
-pub struct Libmdbx(Database);
+pub struct Libmdbx<'a, T: Table> {
```

Comment:
We will be importing this type outside the trie module, so I think we should use a less generic name for it, maybe something like `LibmdbxTrieDB`

Reply:
Good idea! Done.